auth_app/routers/tokens.py

```diff
@@ -0,0 +1,187 @@
+from typing import Annotated, Optional
+
+from fastapi import (
+    APIRouter,
+    Body,
+    Depends,
+    HTTPException,
+    Security,
+    status,
+)
+from fastapi.security import HTTPAuthorizationCredentials
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from auth_app.config import settings
+from auth_app.db.connect_db import get_db
+from auth_app.repositories.tokens import TokenRepo
+from auth_app.repositories.users import UserRepo
+from auth_app.routers.users import get_repository as get_user_repo
+from auth_app.schemes.tokens import (
+    AuthData,
+    CreateData,
+    CreateRefresh,
+    GetAccess,
+    GetRefresh,
+    UpdateRefresh,
+)
+from auth_app.schemes.users import RoleEnum
+from auth_app.services.tokens.authenticate_user import authenticate_user
+from auth_app.services.tokens.general import (
+    requre_expired,
+    verify_refresh,
+)
+from auth_app.services.tokens.jwt import (
+    generate_access,
+    generate_refresh,
+)
+from auth_app.services.tokens.security import oauth2_scheme
+
+token_router = APIRouter(
+    prefix='/tokens',
+    tags=['tokens'],
+)
+
+
+# flake8: noqa: B008
+async def get_repository(session: AsyncSession = Depends(get_db)) -> TokenRepo:
+    return TokenRepo(session)
+
+
+@token_router.post(
+    path='/refresh/get',
+    response_model=GetRefresh,
+    description='Get refresh token for the user',
+    status_code=status.HTTP_200_OK,
+)
+async def get_refresh(
+    auth_data: Annotated[AuthData, Body()],
+    repo: TokenRepo = Depends(get_repository),
+    user_repo: UserRepo = Depends(get_user_repo),
+) -> Optional[GetRefresh]:
+    user = await authenticate_user(
+        **auth_data.model_dump(),
+        user_repo=user_repo,
+    )
+    if not user:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail='User not found or Invalid user data',
+        )
+    result = await repo.get_refresh(
+        user_id=user.id,
+    )
+    return result
+
+
+@token_router.post(
+    path='/refresh/create',
+    response_model=GetRefresh,
+    description='Generate refresh token for the user',
+    status_code=status.HTTP_201_CREATED,
+)
+async def create_refresh(
+    auth_data: Annotated[AuthData, Body()],
+    role: str = RoleEnum.USER,
+    admin_secret: Optional[str] = None,
+    repo: TokenRepo = Depends(get_repository),
+    user_repo: UserRepo = Depends(get_user_repo),
+) -> GetRefresh:
+    user = await authenticate_user(
+        **auth_data.model_dump(),
+        user_repo=user_repo,
+    )
+    if not user:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail='User not found or Invalid user data',
+        )
+    create_data = CreateData.model_validate(
+        {
+            "user_id": str(user.id),
+            "email": user.email,
+            "role": role,
+            "admin_secret": admin_secret,
+        }
+    )
+    token_data = generate_refresh(create_data)
```

Comment:
Не очень очень понял зачем тут такая инициализация. Вот более простой вариант. А главное легко читаемый)
```suggestion
create_data = CreateData(
    user_id=str(user.id),
    email=user.email,
    role=role,
    admin_secret=admin_secret,
)
```

Reply:
исправлено